src/test/cs/Infinispan/HotRod/RemoteFulltextQueryTest.cs

```diff
@@ -0,0 +1,386 @@
+using NUnit.Framework;
+
+/**
+ * This is a copy of QueryStringTest.java, modified for C#.
+ * 
+ */
+namespace Infinispan.HotRod.Tests
+{
+    class RemoteFullTextQueryTest
+    {
+        RemoteCacheManager remoteManager;
+        const String ERRORS_KEY_SUFFIX = ".errors";
+        const String PROTOBUF_METADATA_CACHE_NAME = "___protobuf_metadata";
+        const String NAMED_CACHE = "InMemoryNonSharedIndexFullText";
+
+        [TestFixtureSetUp]
+        public void BeforeClass()
+        {
+            ConfigurationBuilder conf = new ConfigurationBuilder();
+            conf.AddServer().Host("127.0.0.1").Port(11222);
+            conf.ConnectionTimeout(90000).SocketTimeout(6000);
+            conf.Marshaller(new BasicTypesProtoStreamMarshaller());
+            remoteManager = new RemoteCacheManager(conf.Build(), true);
+
+            IRemoteCache<String, String> metadataCache = remoteManager.GetCache<String, String>(PROTOBUF_METADATA_CACHE_NAME);
+            metadataCache.Put("sample_bank_account/bank.proto", File.ReadAllText("proto2/bank.proto"));
+            if (metadataCache.ContainsKey(ERRORS_KEY_SUFFIX))
+            {
+                Console.WriteLine("fail: error in registering .proto model");
+                Environment.Exit(-1);
+            }
+
+            IRemoteCache<String, Transaction> transactionCache = remoteManager.GetCache<String, Transaction>(NAMED_CACHE);
+            PutTransactions(transactionCache);
+        }
+
+        [Test]
+        public void TestExactMatch()
+        {
+            IRemoteCache<String, Transaction> transactionCache = remoteManager.GetCache<String, Transaction>(NAMED_CACHE);
+
+            QueryRequest qr = new QueryRequest();
+            qr.QueryString = "from sample_bank_account.Transaction t where t.description = 'Birthday present'";
+
+            QueryResponse result = transactionCache.Query(qr);
+
+            List<Transaction> transactions = RemoteQueryUtils.unwrapResults<Transaction>(result);
+            Assert.AreEqual(1, transactions.Count);
+        }
+
+        [Test]
+        public void TestFullTextTerm()
+        {
+            IRemoteCache<String, Transaction> transactionCache = remoteManager.GetCache<String, Transaction>(NAMED_CACHE);
+
+            QueryRequest qr = new QueryRequest();
+            qr.QueryString = "from sample_bank_account.Transaction t where longDescription:'rent'";
+
+            QueryResponse result = transactionCache.Query(qr);
+
+            List<Transaction> transactions = RemoteQueryUtils.unwrapResults<Transaction>(result);
+            Assert.AreEqual(1, transactions.Count);
+        }
+
+        [Test]
+        public void TestFullTextTermRightOperandAnalyzed()
+        {
+            IRemoteCache<String, Transaction> transactionCache = remoteManager.GetCache<String, Transaction>(NAMED_CACHE);
+
+            QueryRequest qr = new QueryRequest();
+            qr.QueryString = "from sample_bank_account.Transaction t where longDescription:'RENT'";
```

Comment:
Keep it upper case to prove it works regardless of the case after fixing ISPN-7301. Lower case was already working before that.